storage/engine: separate ExporToSst's MVCCKey into roachpb.Key and hl…

hueypark opened the pull request:
…c.Timestamp

`roachpb.Key` and `hlc.Timestamp` are not directly related, but were merged into `MVCCKey`. To eliminate the confusion caused by this, separate the two.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42366)
<!-- Reviewable:end -->

petermattis reviewed (approved):
This will conflict with https://github.com/cockroachdb/cockroach/pull/42356. The fixups should be straightforward, though.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42366#-:-LtPgrdk09CxarO6Y0im:bp3w2wh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @hueypark)

___
*[pkg/storage/engine/engine.go, line 195 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/42366#-LtPgcWRDSm-N97QEq3N:-LtPgcWRDSm-N97QEq3O:b-tkm0st) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cb94cf7d8752393b54aabbd57aa4bf00d96a72/pkg/storage/engine/engine.go#L195)):*
> ```Go
> 	Closed() bool
> 	// ExportToSst exports changes to the keyrange [start.Key, end.Key) over the
> 	// interval (start.Timestamp, end.Timestamp]. Passing exportAllRevisions exports
> ```

Can you fix up the comment to reference the correct variables?


<!-- Sent from Reviewable.io -->

hueypark reviewed (commented):
I will resolve this after completion of https://github.com/cockroachdb/cockroach/pull/42356.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42366#-:-LtQ1KNr7eCSBwLAcKSj:b-txczuh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @petermattis)

___
*[pkg/storage/engine/engine.go, line 195 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/42366#-LtPgcWRDSm-N97QEq3N:-LtQ1RTR6_EJf_mSDgNz:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cb94cf7d8752393b54aabbd57aa4bf00d96a72/pkg/storage/engine/engine.go#L195)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary><blockquote>

Can you fix up the comment to reference the correct variables?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

petermattis reviewed (commented):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> I'll kick of the merge once this is rebased. (#42356 has merged).

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42366#-:-LtQMT8J4RIhmwTuhOij:b-v3kg98)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

hueypark committed:
storage/engine: separate ExporToSst's MVCCKey into roachpb.Key and hlc.Timestamp

`roachpb.Key` and `hlc.Timestamp` are not directly related, but were merged into `MVCCKey`. To eliminate the confusion caused by this, separate the two.

Release note: None

hueypark reviewed (commented):
I rebase this.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42366#-:-LtRYPA1EvRXSQ2KTb-c:b-rf5guo)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

petermattis commented:
Thanks!

bors r+
